projects/scannet_offline_eval/build_sparse_voxel_map.py

```diff
@@ -0,0 +1,291 @@
+
+    def show(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Display the aggregated point cloud."""
+        return self.voxel_map.show(
+            instances=True,
+            height=1000,
+            boxes_plot_together=False,
+            boxes_name_int_to_display_name_dict=dict(
+                enumerate(self.metadata.thing_classes)
+            ),
+            backend="pytorch3d",
+        )
+
+
+# TODO: use this code to create semantic sensor
+# def create_semantic_sensor(
+#     device_id: int = 0,
+#     verbose: bool = True,
+#     **kwargs,
+# ):
+#     """Create segmentation sensor and load config. Returns config from file, as well as a OvmmPerception object that can be used to label scenes."""
+#     print("- Loading configuration")
+#     config = load_config(visualize=False, **kwargs)
+
+#     print("- Create and load vocabulary and perception model")
+#     semantic_sensor = OvmmPerception(config, device_id, verbose, module="detic")
+#     obj_name_to_id, rec_name_to_id = read_category_map_file(
+#         config.ENVIRONMENT.category_map_file
+#     )
+#     vocab = build_vocab_from_category_map(obj_name_to_id, rec_name_to_id)
+#     semantic_sensor.update_vocabulary_list(vocab, 0)
+#     semantic_sensor.set_vocabulary(0)
+#     return config, semantic_sensor
+
+
+if __name__ == "__main__":
+    from eval.utils import eval_bboxes_and_print
+
+    parser = argparse.ArgumentParser(
+        prog="build_3d_map",
+        description="Builds 3D map of a scannet scene from a RGBD trajectory and evaluates object detection",
+    )
+
+    parser.add_argument(
+        "--scannet-dir",
+        default="/private/home/ssax/home-robot/src/home_robot/home_robot/datasets/scannet/data",
+        help="Directory where scannet dataset lives",
+    )
+    parser.add_argument(
+        "-f",
+        "--frame-skip",
+        type=int,
+        default=180,
+        help="Subsample every frame-skip frames in the trajectory",
+    )
+    parser.add_argument(
+        "-s",
+        "--scene-name",
+        type=str,
+        default="scene0192_00",
+        help="Which ScanNet scene to load. scene0192_00 is a small scene, scene0000_00 is a large scene",
+    )
+    parser.add_argument(
+        "--detector-config",
+        type=str,
+        default=None,
+        help="Location of detector config file",
+    )
+    parser.add_argument(
+        "-d",
+        "--device",
+        type=int,
+        default=0,
+        help="Which device to use for torch operations. -1 for cpu",
+    )
+    parser.add_argument(
+        "--show-open3d",
+        action="store_true",
+        help="Display the scene in an Open3D visualizer window",
+    )
+    args = parser.parse_args()
+    torch_device = "cpu" if args.device == -1 else f"cuda:{args.device}"
+
+    # Load specific scene
+    data = ScanNetDataset(
+        root_dir=args.scannet_dir,
+        frame_skip=args.frame_skip,
+        # referit3d_config = ReferIt3dDataConfig(),
```

Comment:
Dead code or add a flag maybe?

Reply:
maybe htis is in config somewhere?

Reply:
Moved to hydra config -- will push changes soon